src/main/java/software/amazon/awssdk/crt/io/ClientBootstrap.java

```diff
@@ -25,6 +25,14 @@
      * or if the system is unable to allocate space for a native client bootstrap object
      */
     public ClientBootstrap(EventLoopGroup elg, HostResolver hr) throws CrtRuntimeException {
+        if (elg == null) {
```

Comment:
It would be better to keep this constructor's requirements to have a non-null eventLoopGroup and hostResolver and add an overload with no parameters use the static.